### PR TITLE
feat: 🎸 Add Third button to SQFormDialog

### PR DESCRIPTION
**Pull request by aharpt**
Added third button functionality to SQFormDialog

https://www.loom.com/share/0619324e66cd4f2ca80f9a034ff8dd89

✅ Closes: #629

### CODE REVIEW

File: stories/__tests__/SQFormDialog.stories.test.js [hunk 1]
```diff
@@ -168,3 +168,40 @@ describe('Tests for WithAutoFocus', () => {
     expect(textfield).toBe(document.activeElement);
   });
 });
+
+describe('Tests for Tertiary Button', () => {
+  it('should find the tertiary button', async () => {
```
Comment (SeanGroff) [inline stories/__tests__/SQFormDialog.stories.test.js:173]: We always want to speak in terms of what the user would see or experience, rather than the test itself.

'Should display the tertiary button'

File: src/components/SQFormDialog/SQFormDialogInner.js [hunk 2]
```diff
@@ -124,17 +127,57 @@ function SQFormDialogInner({
               showSecondaryButton ? actionsClasses : primaryActionsClasses
             }
           >
-            {showSecondaryButton && (
-              <RoundedButton
-                title={cancelButtonText}
-                onClick={handleCancel}
-                color="secondary"
-                variant="outlined"
+            {showTertiaryButton ? (
```
Comment (SeanGroff) [inline src/components/SQFormDialog/SQFormDialogInner.js:130]: We can take a functional approach to avoid these conditionals.
We can create render functions that contain the logic, we can create separate components that handle this logic and except the necessary props, or we can live this as is.

**Review by SeanGroff — APPROVED** [review]

File: src/components/SQFormDialog/SQFormDialogInner.js (hunk 2 above)
Comment (20BBrown14) [reply src/components/SQFormDialog/SQFormDialogInner.js:130]: I'm in the "Create render functions" camp on this one. I found it a little hard to follow what was going on in the diff.

File: stories/__tests__/SQFormDialog.stories.test.js [hunk 3]
```diff
@@ -168,3 +168,40 @@ describe('Tests for WithAutoFocus', () => {
...
+        isOpen={true}
+        onSave={handleSave}
+        onClose={handleClose}
+        showTertiaryButton={true}
+        tertiaryButtonText={tertiaryButtonText}
+        title="Test"
+      />
+    );
+
+    await screen.findByText(tertiaryButtonText);
```
Comment (laurelbean) [inline stories/__tests__/SQFormDialog.stories.test.js:186]: There are no expects in this test, so nothing is actually being tested. Probably want to add a check for if the button is visible and maybe with the expected button text.

Comment (aharpt) [reply stories/__tests__/SQFormDialog.stories.test.js]: @laurelbean I added the tests, but if you could double-check them I would appreciate it.  Pretty new to writing tests...

Comment (laurelbean) [reply stories/__tests__/SQFormDialog.stories.test.js]: It's looking better!

The 2 lines below are essentially doing the same thing, checking for the existence of the button. So you don't need both of them. Also, unless you're expecting to wait for loading or an animation, you don't need to `waitFor` an expect. 

```
expect(
   screen.getByRole('button', {name: tertiaryButtonText})
).toBeInTheDocument();

await waitFor(() => {
    expect(screen.getByText(tertiaryButtonText)).toBeInTheDocument();
});```

If you wanted to check for the text of the button, then I'd save the `getByRole` to a variable, and check that for existence and for text. Something like:
```
const tertiaryButton = screen.getByRole('button', {name: tertiaryButtonText});
expect(tertiaryButton).toBeInTheDocument();

expect(tertiaryButton).toHaveTextContext(tertiaryButtonText);
or 
expect(tertiaryButton).toHaveText(tertiaryButtonText);
```

Comment (aharpt) [reply stories/__tests__/SQFormDialog.stories.test.js]: @laurelbean Thanks for helping me out with this!  The correct changes should now be part of this PR.  Let me know if anything further needs to be addressed.

**Review by 20BBrown14 — APPROVED** [review]
